### PR TITLE
Provide soft config to hide General > Users tab in Admin console

### LINKED ISSUE #2946: Provide option to hide the Admin > General > Users tab
The Admin console ships with a default, minimal, read-only users tab, nested under the General top-level tab.

For applications that need to provide more advanced user management within the admin console - e.g. adding and editing users - this built in tab can be confusing. In other scenarios, we might be loading users lazily from an API or other auth workflow, meaning this tab might not offer a complete listing of users and is misleading in another wya.

It would be good to provide an option to disable it. hiding it from the admin interface.

### PR DESCRIPTION
**Pull request by saba-mo**
Resolves #2946 

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: admin/tabs/general/GeneralTab.js [hunk 1]
```diff
@@ -21,9 +21,14 @@ export const generalTab = hoistCmp.factory(
...
-                {id: 'users', icon: Icon.users(), content: userPanel},
+                {id: 'users', icon: Icon.users(), content: userPanel, omit: omitTab('hideUsersTab')},
                 {id: 'alertBanner', icon: Icon.bullhorn(), content: alertBannerPanel}
             ]
         }
     })
 );
+
+const omitTab = (field) => {
+    const conf = XH.getConf('xhAdminConsoleConfig', {});
```
Comment (lbwexler) [inline admin/tabs/general/GeneralTab.js:32]: Can we call it `xhAdminAppConfig`?  Can we add a default value in hoist core, so easy for new app users to find it?   

Also this function should just be called `hideUsersTab()` doesn't seem like it is ever going to be a general tab hiding mechanism

Comment (lbwexler) [inline admin/tabs/general/GeneralTab.js:32]: Can we call it `xhAdminAppConfig`?  Can we add a default value in hoist core, so easy for new app users to find it?   

Also this function should just be called `hideUsersTab()` doesn't seem like it is ever going to be a general tab hiding mechanism

Comment (amcclain) [reply admin/tabs/general/GeneralTab.js]: We discussed the merits of adding a bootstrapped value in hoist-core.  I voted against it for the moment, while we only have one little flag here, and a rarely used one at that. 

Don't feel strongly about it, but in general I don't mind the pattern of supporting some more rare edge-case configs with baked-in defaults and keeping the list in most apps a bit more easily scannable.

Agree re. `omitTab` - a pattern I've used in a few places is to have a getter on my *model* class like:

`get conf() {return XH.getConf('someJsonConfigForMyModel', {})}` 

and then in the component here you could go with `omit: model.conf.hideUsersTab`

Comment (amcclain) [reply admin/tabs/general/GeneralTab.js]: Woops - got caught up in another convo and submitted this too late - thanks for merging and getting this change in!

**Pull request merged**